src/tsickle.ts

```diff
@@ -888,6 +888,36 @@ class ExternsWriter extends ClosureRewriter {
       this.writeExternsFunction(name.getText(), paramNames, namespace);
     }
 
+    // Handle method declarations/signatures first, since we need to deal with overloads.
+    // TODO: Build a collection of the remaining members
+    let methods: {[id: string]: ts.MethodDeclaration[];} = {};
+    for (let member of decl.members) {
+      if (member.kind === ts.SyntaxKind.MethodSignature ||
+          member.kind === ts.SyntaxKind.MethodDeclaration) {
+        let m = <ts.MethodDeclaration>member;
+        let overloaded = methods[m.name.getText()];
+        if (overloaded) {
+          methods[m.name.getText()].push(m);
+        } else {
+          methods[m.name.getText()] = [m];
+        }
+      }
+    }
+    for (let methodName in methods) {
+      if (methods.hasOwnProperty(methodName)) {
+        let method: ts.MethodDeclaration[] = methods[methodName];
+        let rootMethod = method[0];
+        if (method.length > 1) {
+          this.debugWarn(method[0], 'overloaded method signatures found');
```

Comment:
Another option (your call) is to
```js
this.emit(`/* TODO: method overloaded */\n`);
```
as that shows up in the golden file and is obvious to humans.